corehq/apps/es/transient_util.py

```diff
@@ -73,6 +73,19 @@ def index_info_from_cname(cname):
     return CANONICAL_NAME_INFO_MAP[cname]
 
 
+def index_info_from_adapter(adapter):
```

Comment:
Adding a new utility to map from an adapter back to an index info object feels a bit backwards. How about changing the functions that still need info objects to support both? This causes an issue with the `assume_alias()` function because adapters know nothing about index aliases, but that is fine -- creating aliases on the Elasticsearch cluster itself can safely be dropped since aliases haven't been used for anything other than adapter lookups in code for quite some time now.

See 689c434715f94ecdab91769ccc78ccd5bc8b94a3 for a full proposal.

Reply:
@AmitPhulera Is this suggestion expected to be implemented in this PR?

Reply:
I looked into it and supporting both would require bringing back support for ES Info objects in the ESProcesser. 
So in this PR I plan to keep it as it is. I have added a TODO in the code which I plan to do with the final cleanups that are planned after this PR is merged.